[WS-7] Code quality improvement

gohanko opened the pull request:
This pull request is a follow up to #3.

### Changelog
- Moved all react components into .jsx files of their own.
- Added proptypes to components and removed some unused code.
- Move all paths constants into an object.
- Added ESLint.

gohanko commented:
@akiyamasho I noticed that on `/cast` we have everyone who worked on the anime. Wouldn't it be better to do `/credits` instead?

gohanko commented:
@akiyamasho If we decide to keep `/cast` then this pr is ready to be merged.

akiyamasho commented:
@gohanko Sorry for the late review!

Yes I think it makes sense to convert it to `/credits` but keep the route to `/cast` for backwards compatibility, since we may have posted links to `/cast` on old social media posts.

Other than that, I think the changes look great.

akiyamasho commented on src/Constants/paths.js:
```diff
@@ -0,0 +1,8 @@
+const paths = {
+    about: '/about',
+    staff: '/cast',
```
As discussed in previous comment:

> Yes I think it makes sense to convert it to /credits but keep the route to /cast for backwards compatibility, since we may have posted links to /cast on old social media posts.

gohanko commented:
@akiyamasho I renamed it to `cast` to `credits` and `/cast` now redirects to `/credits`.

I also changed the wording on the navigation bar. Is `クレジット` the correct translation for credits in this context?

akiyamasho commented:
@gohanko Thanks! スタッフ・キャスト would be more apropriate

gohanko commented:
@akiyamasho Updated both `en` and `ja`.

akiyamasho commented:
LGTM. Great work!
